packages/runtime/test/transport/relationshipTemplates.test.ts

```diff
@@ -327,4 +327,68 @@ describe("RelationshipTemplates query", () => {
     });
+
+    test("password query", async () => {
+        const passwordProtectedTemplate = await createTemplate(runtimeServices1.transport, undefined, { password: "password" });
+
+        const pinProtectedTemplate = await createTemplate(runtimeServices1.transport, undefined, { password: "1234", passwordIsPin: true });
+
+        const passwordProtectedTemplates = (
+            await runtimeServices1.transport.relationshipTemplates.getRelationshipTemplates({
+                query: {
+                    "passwordProtection.password": "password"
+                }
+            })
+        ).value;
+        const passwordProtectedTemplateIds = passwordProtectedTemplates.map((t) => t.id);
+
+        expect(passwordProtectedTemplateIds).toContain(passwordProtectedTemplate.id);
+        expect(passwordProtectedTemplateIds).not.toContain(pinProtectedTemplate.id);
+
+        const pinProtectedTemplates = (
+            await runtimeServices1.transport.relationshipTemplates.getRelationshipTemplates({
+                query: {
+                    "passwordProtection.password": "1234"
+                }
+            })
+        ).value;
+        const pinProtectedTemplateIds = pinProtectedTemplates.map((t) => t.id);
+
+        expect(pinProtectedTemplateIds).not.toContain(passwordProtectedTemplate.id);
+        expect(pinProtectedTemplateIds).toContain(pinProtectedTemplate.id);
+    });
+
+    test("passwordIsPin query", async () => {
+        const passwordProtectedTemplate = await createTemplate(runtimeServices1.transport, undefined, { password: "password" });
+
+        const pinProtectedTemplate = await createTemplate(runtimeServices1.transport, undefined, { password: "1234", passwordIsPin: true });
+
+        const unprotectedTemplate = await createTemplate(runtimeServices1.transport);
+
+        const templatesWithoutPin = (
+            await runtimeServices1.transport.relationshipTemplates.getRelationshipTemplates({
+                query: {
+                    "passwordProtection.passwordIsPin": "!true"
+                }
+            })
+        ).value;
+        const templateIdsWithoutPin = templatesWithoutPin.map((t) => t.id);
+
+        expect(templateIdsWithoutPin).toContain(passwordProtectedTemplate.id);
+        expect(templateIdsWithoutPin).not.toContain(pinProtectedTemplate.id);
+        expect(templateIdsWithoutPin).toContain(unprotectedTemplate.id);
+
+        const templatesWithPin = (
+            await runtimeServices1.transport.relationshipTemplates.getRelationshipTemplates({
+                query: {
+                    "passwordProtection.passwordIsPin": "true"
+                }
+            })
+        ).value;
+        const templateIdsWithPin = templatesWithPin.map((t) => t.id);
+
+        expect(templateIdsWithPin).not.toContain(passwordProtectedTemplate.id);
+        expect(templateIdsWithPin).toContain(pinProtectedTemplate.id);
+        expect(templateIdsWithPin).not.toContain(unprotectedTemplate.id);
+    });
```

Comment:
please see the other tests that do querying, we have a fabulous tester for querystrings :)